go/runtime/host/sgx/tcb_cache.go

```diff
@@ -0,0 +1,88 @@
+package sgx
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/persistent"
+	"github.com/oasisprotocol/oasis-core/go/common/sgx/pcs"
+)
+
+const (
+	tcbCacheKey = "tcb_bundle_cache"
+
+	tcbCacheRefreshThreshold    = 14 * 24 * time.Hour
+	tcbCacheSlowRefreshInterval = 24 * time.Hour
+)
+
+type tcbBundleCache struct {
+	bundle         *pcs.TCBBundle `json:"bundle"`
+	expectedExpiry time.Time      `json:"expected_expiry"`
+	lastUpdate     time.Time      `json:"last_update"`
```

Comment:
I believe this will not work (e.g. unmarshalling will just zero-initialize these) as private fields cannot be deserialized properly (so you need to capitalize them to export).

Would be good to have some basic tests of the cache that makes sure storage/retrieval works. To make testing easier you could create a structure (e.g. `tcbCache`) that one can create and holds a service store and logger. Then in tests you can more easily (e.g. using a test service store) test it without the need to create an `sgxProvisioner`.